Drop net6.0 target

mus65 opened the pull request:
.NET 6 is EOL since last November. Also simplified a lot of .NET version specific conditions in the process.

mus65 committed:
Drop net6.0 target

mus65 commented:
Test failure is unrelated, see #1253

Rob-Hague commented on Directory.Build.props:
```diff
@@ -31,7 +31,7 @@
         Disable nullable warnings on old frameworks because of missing annotations.
   -->
-  <PropertyGroup Condition=" !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net6.0')) ">
+  <PropertyGroup Condition=" !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0')) ">
```
```suggestion
  <PropertyGroup Condition=" '$(TargetFrameworkIdentifier)' != '.NETCoreApp' ">
```

mus65 replied:
This doesn't actually work. Seems like this is not defined yet in Directory.Build.props, it does work in Directory.Build.targets though, moved it there. See also https://github.com/dotnet/runtime/blob/4bd597ffde128555b4ff017e87c60adc2fedd178/src/libraries/Directory.Build.targets#L22 .

Rob-Hague replied:
huh, it looked like it worked locally before I suggested it 🤷

mus65 replied:
Did you actually try to trigger a nullable warning? Building works fine with your suggestion, but is disables the warnings for modern .NET as well.

Rob-Hague replied:
yeah, I inverted the condition initially to test. No matter. I will check this PR is OK with the others before merging

Rob-Hague commented on src/Renci.SshNet/Common/TaskToAsyncResult.cs:
```diff
@@ -1,5 +1,5 @@
 ﻿#pragma warning disable
-#if !NET8_0_OR_GREATER
+#if !NET
 // Copied verbatim from https://github.com/dotnet/runtime/blob/78bd7debe6d8b454294c673c9cb969c6b8a14692/src/libraries/Common/src/System/Threading/Tasks/TaskToAsyncResult.cs
```
```suggestion
// Copied verbatim from https://github.com/dotnet/runtime/blob/261611930d6b436d7c4395450356b624d903d9bf/src/libraries/Common/src/System/Threading/Tasks/TaskToAsyncResult.cs
```

Rob-Hague commented on src/Renci.SshNet/Security/Cryptography/Ciphers/AesGcmCipher.BclImpl.cs:
```diff
@@ -1,4 +1,4 @@
-#if NET6_0_OR_GREATER
+#if NET
```
the remaining ifdefs in this file can be removed

mus65 committed:
Update src/Renci.SshNet/Common/TaskToAsyncResult.cs

Co-authored-by: Rob Hague <rob.hague00@gmail.com>

mus65 committed:
remove redundant #if

for some reason this made the compiler suddenly
realize that the plain text variables are unused.

mus65 committed:
use TargetFrameworkIdentifier

this doesn't work in Directory.Build.props, moved it to Directory.Build.targets.

mus65 committed:
fix null reference warnings in Benchmarks

seems like the warnings were (somehow) disabled here
before and were fixed by the previous TargetFrameworkIdentifier
change.

mus65 committed:
Merge branch 'develop' into dropnet6

Rob-Hague committed:
fix unused plainTextOffset in AesGcmCipher.BclImpl

mus65 committed:
Merge remote-tracking branch 'upstream/develop' into dropnet6

mus65 committed:
CI retry

Rob-Hague committed:
more cosmetics

Rob-Hague committed:
more

Rob-Hague commented on src/Renci.SshNet/Renci.SshNet.csproj:
```diff
@@ -36,11 +36,7 @@
-  </PropertyGroup>
-  
-  <PropertyGroup Condition=" $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0')) ">
+  <PropertyGroup Condition=" '$(TargetFrameworkIdentifier)' == '.NETCoreApp' ">
```
I am actually not sure now that this works. I will do some tests and if I can't convince myself then I will revert the TargetFrameworkIdentifier changes. Not sure where I got that from

Rob-Hague replied:
The following patch successfully fails the AotCompatibilityTestApp publish on both branches, but on `develop` I also get a diagnostic in VS which does not appear on this branch. So I will revert those changes. Sorry for the noise

```diff
diff --git a/src/Renci.SshNet/SshCommand.cs b/src/Renci.SshNet/SshCommand.cs
index f900e843..9bcafd85 100644
--- a/src/Renci.SshNet/SshCommand.cs
+++ b/src/Renci.SshNet/SshCommand.cs
@@ -214,6 +214,8 @@ public string Error
         /// <exception cref="ArgumentNullException">Either <paramref name="session"/>, <paramref name="commandText"/> is <see langword="null"/>.</exception>
         internal SshCommand(ISession session, string commandText, Encoding encoding)
         {
+            _ = Activator.CreateInstance(session.GetType());
+
             ThrowHelper.ThrowIfNull(session);
             ThrowHelper.ThrowIfNull(commandText);
             ThrowHelper.ThrowIfNull(encoding);
```

Rob-Hague committed:
update README

Rob-Hague committed:
Revert "use TargetFrameworkIdentifier"

This reverts commit 076ede161d05a1418710c68ba7ecc647bc74ef5c.

Rob-Hague committed:
Merge branch 'develop' into dropnet6
